src/providers/provider.ts

```diff
@@ -17,13 +18,31 @@ export class Provider {
   protected _eventManager: EventManager;
   protected _logger: Logger;
   public cuePointManager: CuePointManager | null = null;
+  private _dataCollector: DataCollector | null = null;
```

Comment:
from the name "data collector" i dont intuitively understand what the util does. how about something with "cue point" and "history" / "store" / "timed-store" / "store-with-expiration" because thats kind of what it does ?

Reply:
TimedDataCollector